extensions/mapstore/src/main/java/com/hazelcast/mapstore/MappingHelper.java

```diff
@@ -33,17 +36,57 @@ final class MappingHelper {
 
     public void createMapping(String mappingName, String tableName, String mappingColumns,
                               String dataConnectionRef, String idColumn) {
+
         sqlService.execute(
                 "CREATE MAPPING \"" + mappingName + "\" "
-                + "EXTERNAL NAME \"" + tableName + "\" "
-                + (mappingColumns != null ? " ( " + mappingColumns + " ) " : "")
-                + " DATA CONNECTION \"" + dataConnectionRef + "\" "
-                + " OPTIONS ("
-                + "    'idColumn' = '" + idColumn + "' "
-                + ")"
+                        + "EXTERNAL NAME " + externalName(tableName) + " "
+                        + (mappingColumns != null ? " ( " + mappingColumns + " ) " : "")
+                        + " DATA CONNECTION \"" + dataConnectionRef + "\" "
+                        + " OPTIONS ("
+                        + "    'idColumn' = '" + idColumn + "' "
+                        + ")"
         ).close();
     }
 
+    //package-private just for testing
+    static String externalName(String tableName) {
+        return splitByNonQuotedDots(tableName).stream()
+                .map(unwrapFromQuotesIfPresent())
+                .map(wrapWithQuotes())
+                .collect(Collectors.joining("."));
+    }
+
+    private static List<String> splitByNonQuotedDots(String input) {
+        List<String> result = new ArrayList<>();
+        int tokenStart = 0;
+        boolean inQuotes = false;
+        for (int i = 0; i < input.length(); i++) {
+            switch (input.charAt(i)) {
+                case '\"':
```

Comment:
`[]` should be also taken into consideration for MSSQL. 

For more general, use `SqlDialect`'s  `DEFAULT_CONTEXT.identifierQuoteString()` value

Reply:
We agreed with @frant-hartm that we should use only double quotes as parts delimators: https://github.com/hazelcast/hazelcast/pull/24332/commits/7772527f855573a1171b351f771cb638c23a2e2f

Reply:
> Do you assume user here must use " only, instead of MSSQL syntax for quoted name?

Yes, this how I understood @frant-hartm

Reply:
Yes, I think it's best to keep consistent with how the JDBC mapping will be created, which is `"database.name"."tableName"`. It will also show like that in the `information_schema.mappings` table and GET DDL definition for the created `__map-store.mapping`.

The JDBC connector then will translate this internally into `[database.name].tableName` for MSSQL.